e2e/screenshot.ts

```diff
@@ -0,0 +1,52 @@
+import * as fs from 'fs';
+import * as gulp from 'gulp';
+import * as path from 'path';
+import {browser} from 'protractor';
+
+export class Screenshot {
+  id: string;
+
+  /**
+   * The filename used to store the screenshot
+   * @returns {string}
+   */
+  get filename() {
```

Comment:
Can do `get filename(): string` and remove the type from the JsDoc
(here and elsewhere for return types and params)